bin.src/translate_header.py

```diff
@@ -14,18 +19,56 @@ def read_metadata(file, hdu=1):
         fits_file = fits.open(file)
         return fits_file[hdu].header
 
+parser = argparse.ArgumentParser(description="Summarize headers from astronomical data files")
+parser.add_argument("files", metavar="file", type=str, nargs="+",
+                    help="File(s) from which headers will be parsed."
+                    " If a directory is given it will be scanned for files matching the regular"
+                    " expression defined in --regex.")
+parser.add_argument("-q", "--quiet", const=True, default=False, action="store_const",
+                    help="Do not report the translation content from each header")
+parser.add_argument("-d", "--dumphdr", const=True, default=False, action="store_const",
+                    help="Dump the header in YAML format to standard output rather than translating it")
+parser.add_argument("--traceback", const=True, default=False, action="store_const",
+                    help="Give detailed trace back when any errors encountered")
 
-failed = []
-for file in sys.argv[1:]:
-    print(f"{file}:")
-    md = read_metadata(file)
+re_default = r"\.fit[s]?\b"
```

Comment:
The entire package is written with underscore naming.  We made this decision early on because the translation methods `to_datetime_begin` mapping to butler gen 3 datetime_begin was a lot more obvious than using `toDatetimeBegin`. Also we hope to make the package astropy affiliated and thought it best to be more standard with the rest of the community.  The public interface (ObservationInfo is very thin).